frontend/components/SearchComponent.tsx

```diff
@@ -0,0 +1,185 @@
+import Link from 'next/link';
+import { useState } from 'react';
+import { ArtifactSearchParams } from '@/util/searchParams';
+import { Search, SquareX } from 'tabler-icons-react';
+
+interface SearchComponentProps {
+  resourceType: ArtifactResourceType;
+}
+
+interface Parameter {
+  name: string;
+  description: string;
+  value: string;
+  date?: Date | null;
+}
+
+/**
+ * SearchComponent is a component for displaying search inputs for a resource
+ */
+export default function SearchComponent({ resourceType }: SearchComponentProps) {
+  const emptyInputs = ArtifactSearchParams[resourceType].map(p => ({
+    name: p.param,
+    description: p.description,
+    value: ''
+  }));
+  const [searchInputs, setSearchInputs] = useState<Parameter[]>(emptyInputs);
+
+  function changeSearchInputs(name: string, newValue: string, newDate?: Date | null) {
+    setSearchInputs(
+      searchInputs?.map(si => {
+        if (si.name === name) {
+          return { ...si, value: newValue, date: newDate };
+        } else {
+          return si;
+        }
+      })
+    );
+  }
+
+  /**
+   * Returns each of the given resource type's search parameters as a Text input or
+   * a date input
+   */
+  const searchInputDisplays = searchInputs.map(si => {
+    if (si.name === 'version') {
+      return (
+        <Grid.Col span={6} key={si.name}>
+          <TextInput
+            label={si.name}
+            value={si.value}
+            description={si.description}
+            disabled={searchInputs.find(si => si.name === 'url')?.value === ''}
+            onChange={event => changeSearchInputs(si.name, event.currentTarget.value)}
+          />
+        </Grid.Col>
+      );
+    } else if (si.name !== 'date') {
+      return (
+        <Grid.Col span={6} key={si.name}>
+          <TextInput
+            label={si.name}
+            value={si.value}
+            description={si.description}
+            onChange={event => changeSearchInputs(si.name, event.currentTarget.value)}
+          />
+        </Grid.Col>
+      );
+    } else {
+      return (
+        <Grid.Col span={6} key={si.name}>
+          <DateInput
+            label={si.name}
+            value={si.date}
+            description={si.description}
+            onChange={event => changeSearchInputs(si.name, si.value, event)}
+          />
+        </Grid.Col>
+      );
+    }
+  });
+
+  /**
+   * Returns the requestPreview string based on the search inputs
+   */
+  const requestPreview = () => {
+    let requestPreview = '';
+    const requestParams: Parameter[] = [];
+    let urlAndVersion: string;
+    searchInputs.forEach(si => {
+      if (si.name === 'url') {
+        urlAndVersion = encodeURIComponent(si.value);
+      } else if (si.name === 'version') {
+        if (urlAndVersion !== '') {
+          si.value !== ''
+            ? requestParams.push(
+                { name: 'url', description: si.description, value: encodeURIComponent(urlAndVersion) },
```

Comment:
```suggestion
                { name: 'url', description: si.description, value: urlAndVersion },
```
This actually re-encodes the `urlAndVersion` already encoded above